### PR TITLE
Verify that the mandatory fields (testPlan, revertPlan)  are non empty when doing arc diff.

### PR DESCRIPTION
**Pull request by raviagarwal7**
- specify mandatory field in .arcconfig as  `"differential.mandatory_fields": ["testPlan", "revertPlan"]`
- Throws an exception if not specified and exits the workflow.

### CODE REVIEW

**Comment by Qix-** [comment]
In a lot of scenarios, Test Plan and Revert Plan are not really relevant. For a simple typo diff, for example, there's no reason to do a test other than with jenkins, which will be there anyway if it's configured with harbormaster.

Same thing with a revert plan - they're not _always_ needed as the workflow for testing/reverting a lot of diffs (at least, with smaller projects) is implicit.

This is an incredibly opinionated PR.

**Comment by raviagarwal7** [comment]
On mobile repos we mandate having "Test Plan" and "Revert Plan" ([example arc config](https://code.uberinternal.com/diffusion/MOANDRO/browse/master/.arcconfig;7556fa510f1417ba02259c854152134a63e07f0e$6)) by using the [UberTestPlanTestEngine](https://github.com/uber/arcanist/blob/master/src/unit/engine/UberTestPlanTestEngine.php) and [UberRevertPlanTestEngine](https://github.com/uber/arcanist/blob/master/src/unit/engine/UberRevertPlanTestEngine.php). 

The issue is that people sometimes by pass these checks with --no-lint and see errors on subsequent diffs. Also, test plan and revert plan test engine uses a very crude way of figuring out whether the values are empty or not. (like, append all the commit messages with the edit-messages.json contents and check if the fields have non-empty value). 

Arcanist diff workflow already parses the message correctly for all scenarios (diff creation, diff update, diff edit message), which makes it easier to do the check here.

We already have reviewers empty check here, which is why I added checks for more fields. (also the check will only be done if `differential.mandatory_fields` is specified in `.arcconfig`.

What would you suggest, that is a good way of doing this check?

File: src/workflow/ArcanistDiffWorkflow.php [hunk 1]
```diff
@@ -1861,6 +1861,29 @@ private function validateCommitMessage(
       }
     }
 
+    $config = $this->getConfigurationManager();
+    $mandatory_fields = $config->getConfigFromAnySource(
```
Comment (Qix-) [inline src/workflow/ArcanistDiffWorkflow.php]: Is this an `.arcconfig` source, or is this coming from somewhere else? Do individual repositories have the ability to turn this off?

If so, then :+1:.

Comment (raviagarwal7) [inline src/workflow/ArcanistDiffWorkflow.php]: Yes, it is an .arcconfig source. The check is off by default. Individual repositories can 
 - enable it by adding `"differential.mandatory_fields": ["testPlan", "revertPlan"]` to arcconfig.
 - disable it (if enabled gloably) by adding `"differential.mandatory_fields": []` to arcconfig.

Comment (Qix-) [inline src/workflow/ArcanistDiffWorkflow.php]: :+1:

**Comment by kageiit** [comment]
@Qix- Updated it to make it more configurable. No functional changes otherwise
